tensorflow/python/lib/io/file_io_test.py

```diff
@@ -582,5 +584,31 @@ def testFileCrc32Binary(self):
     self.assertTrue(crc1 != crc2)
     self.assertEqual(crc2, crc3)
 
+  def testMatchingFilesPermission(self):
+    # Test case for GitHub issue 19274.
+    # Create top level directory test_dir.
+    dir_path = os.path.join(self._base_dir, "test_dir")
+    file_io.create_dir(dir_path)
+    # Create second level directories `noread` and `any`.
+    noread_path = os.path.join(dir_path, "noread")
+    file_io.create_dir(noread_path)
+    any_path = os.path.join(dir_path, "any")
+    file_io.create_dir(any_path)
+    files = ["file1.txt", "file2.txt", "file3.txt"]
+    for name in files:
+      file_path = os.path.join(any_path, name)
+      file_io.FileIO(file_path, mode="w").write("testing")
+    file_path = os.path.join(noread_path, "file4.txt")
+    file_io.FileIO(file_path, mode="w").write("testing")
+    # Change noread to noread access.
+    os.chmod(noread_path, 0)
+    expected_match = [compat.as_bytes(dir_path)]
+    with self.test_session() as sess:
+      self.assertItemsEqual(
+          gen_io_ops.matching_files(dir_path).eval(), expected_match)
```

Comment:
why use gen_io_ops.matching_files.. why not file_io.get_matching_files instead? In fact try looking for .../test_dir/*/file*.txt and expect to only find file1.txt, file2.txt and file3.txt